include/eeprom/core/write_protect.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+namespace eeprom {
+namespace write_protect {
+
+/**
+ * Interface to write protect pin. Must be implemented in FW and Simulation
+ */
+class WriteProtectPin {
```

Comment:
nit: can we separate the `WriteProtectPin` class and call it something like `EepromHardwareIface` (following the pattern of `motor_hardware_interface.hpp`?

Reply:
Ok